spec/unit/provider/package/pkg_spec.rb

```diff
@@ -286,13 +286,7 @@ def self.it_should_respond_to(*actions)
 
           it "should install specific version(2)" do
             resource[:ensure] = '0.0.8'
-            expect(provider).to receive(:properties).and_return({:mark => :hold})
-            expect(Puppet::Util::Execution).to receive(:execute)
-              .with(['/bin/pkg', 'unfreeze', 'dummy'], {:failonfail => false, :combine => true})
-              .and_return(Puppet::Util::Execution::ProcessOutput.new('', 0))
-            expect(Puppet::Util::Execution).to receive(:execute)
-              .with(['/bin/pkg', 'list', '-Hv', 'dummy'], {:failonfail => false, :combine => true})
-              .and_return(Puppet::Util::Execution::ProcessOutput.new('pkg://foo/dummy@0.0.7,5.11-0.151006:20131230T130000Z  installed -----', 0))
+            expect(provider).to receive(:query).with(no_args).and_return({:ensure => '0.0.7,5.11-0.151006:20131230T130000Z'}).exactly(2).times
```

Comment:
Since `query` is a method on the `provider` object that we're testing, it's better to not stub out its behavior. Instead stub out the commands that the provider executes to update to a newer version.

Reply:
You might need to give me a hand on tis one Josh. Rspec is not my strength and this is what I needed to do to make the test work correctly.

Reply:
OK, does this look better now?

Also, I noticed when I ran `bundle exec rubocop` it only descended the `lib` hierarchy. If I run on `spec/unit/provider/package/pkg_spec.rb`, there are a LOT of things that need fixing, most from quite some time ago. Do you want a `rubocop -a` run over it?

Reply:
or do we not worry about `rubocop` and the sec dir (I see it's excluded in `.noexec.yaml`)?

If that is the case, then I think we are ready.

Reply:
I've resolved your issues and it appears that we don't need to do `rubocop` on the spec hierarchy.